powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/EciesDecryptor.java

```diff
@@ -107,34 +108,54 @@ public EciesDecryptor(EciesEnvelopeKey envelopeKey, byte[] sharedInfo2) {
      * and sharedInfo2 parameter.
      *
      * @param ephemeralPublicKeyBytes Ephemeral public key for ECIES.
+     * @param nonce Nonce required for the future response encryption. The parameter may be {@code null} for a legacy V2 ECIES encryption,
+     *              or if the only purpose of the initialization is to generate parameters, for the remote decryptor.
      * @throws EciesException In case envelope key initialization fails.
      */
-    public void initEnvelopeKey(byte[] ephemeralPublicKeyBytes) throws EciesException {
+    public void initEnvelopeKey(byte[] ephemeralPublicKeyBytes, byte[] nonce) throws EciesException {
         envelopeKey = EciesEnvelopeKey.fromPrivateKey(privateKey, ephemeralPublicKeyBytes, sharedInfo1);
         // Invalidate this decryptor for decryption
         canDecryptData = false;
         canEncryptData = true;
+        // Derive IV for future encryption.
+        ivForEncryption = nonce != null ? envelopeKey.deriveIvForNonce(nonce) : new byte[16];
```

Comment:
I am not sure what is the use case for this. The `initEnvelopeKey` is always called with `null` nonce parameter in PowerAuth sources.

Reply:
OK, I removed this possibility and updated the documentation that encryption is available only for a compatibility reasons.